From 8d9bbc893466da39eea9f5b92dc2797cb7d79e12 Mon Sep 17 00:00:00 2001
From: Shikha Vyaghra <vyaghras@amazon.com>
Date: Wed, 8 Jan 2025 19:42:26 +0000
Subject: [PATCH] makefile: allow to use any kernel arch

The added check results in failing build for x86_64 architecture when we use
`_cross_karch` in make commands in spec file and fails for aarch64 while
using _cross_arch in the make commands.

This check has been added recently and do not allow us to use
Kernel_ARCH we want.

We need different Kernel arch as we do cross builds.

Signed-off-by: Shikha Vyaghra <vyaghras@amazon.com>
---
 kernel-open/Makefile | 10 ----------
 kernel/Makefile      | 10 ----------
 2 files changed, 20 deletions(-)

diff --git a/kernel-open/Makefile b/kernel-open/Makefile
index 72672c2..187f39e 100644
--- a/kernel-open/Makefile
+++ b/kernel-open/Makefile
@@ -80,16 +80,6 @@ else
     )
   endif
 
-  KERNEL_ARCH = $(ARCH)
-
-  ifneq ($(filter $(ARCH),i386 x86_64),)
-    KERNEL_ARCH = x86
-  else
-    ifeq ($(filter $(ARCH),arm64 powerpc),)
-        $(error Unsupported architecture $(ARCH))
-    endif
-  endif
-
   NV_KERNEL_MODULES ?= $(wildcard nvidia nvidia-uvm nvidia-vgpu-vfio nvidia-modeset nvidia-drm nvidia-peermem)
   NV_KERNEL_MODULES := $(filter-out $(NV_EXCLUDE_KERNEL_MODULES), \
                                     $(NV_KERNEL_MODULES))
diff --git a/kernel/Makefile b/kernel/Makefile
index 72672c2..187f39e 100644
--- a/kernel/Makefile
+++ b/kernel/Makefile
@@ -80,16 +80,6 @@ else
     )
   endif
 
-  KERNEL_ARCH = $(ARCH)
-
-  ifneq ($(filter $(ARCH),i386 x86_64),)
-    KERNEL_ARCH = x86
-  else
-    ifeq ($(filter $(ARCH),arm64 powerpc),)
-        $(error Unsupported architecture $(ARCH))
-    endif
-  endif
-
   NV_KERNEL_MODULES ?= $(wildcard nvidia nvidia-uvm nvidia-vgpu-vfio nvidia-modeset nvidia-drm nvidia-peermem)
   NV_KERNEL_MODULES := $(filter-out $(NV_EXCLUDE_KERNEL_MODULES), \
                                     $(NV_KERNEL_MODULES))
-- 
2.40.1

